Lib/http/cookiejar.py

```diff
@@ -1217,7 +1218,7 @@ def path_return_ok(self, path, request):
         return False
 
 def vals_sorted_by_key(adict):
-    keys = sorted(adict.keys())
+    keys = deepcopy(adict).keys()
```

Comment:
The name of the function is now wrong.

Reply:
I'm not familiar with the relevant RFC, but I have a strictly implementation-related comment here: `deepcopy` seems unnecessary since all we use the copy for is to get the keys out of it. And we get the keys only to get values corresponding to those keys, in order (I don't suppose `adict.get(k)` can return anything surprising for those keys?).

Since the order is as-is now and there's no sorting involved we can just return `adict.values()`? And since it's been simplified so much I recommend changing the

```python
value = vals_by_key(mapping)
for obj in values:
```
part in `deepvalues()` to just read

```python
for obj in mapping.values()
```

and dropping the whole `vals_by_key` function.

Reply:
Hi. Thanks for the guidance. I just wanted to fix the issue with the minimum change in the code. I totally agree with the changes and dropping `vals_by_keys` But I was actually thinking about doing this:
```python
for obj in mapping.copy().values()
```
since there is this `clear_session_cookies` [function](https://github.com/python/cpython/blob/2d55aa9e37c9c84f4f6a8135d0326da0bcd8f38b/Lib/http/cookiejar.py#L1716) which use the class iterator and delete the object while iterating then causes a runtime error.

thanks in advance for the recommendations.

Reply:
Oh yeah, that's the way to go then (or `for obj in list(maping.values())` to copy just the list of values, not the whole dictionary, but I don't imagine it's a problem either way).